herbert_core/classes/data_loaders/@loader_nxspe/loader_nxspe.m

```diff
@@ -245,6 +265,90 @@
                 obj = set_data_file_name@a_loader(obj,filename);
             end
         end
+        function instrument = read_instrument_info_(obj)
+            if ~isempty(obj.file_name)
+                filename = obj.file_name;
+            else
+                filename = obj.par_file_name;
+            end
+            if ~isempty(obj.root_nexus_dir)
+                root_dir = obj.root_nexus_dir;
+            else
+                root_dir = find_root_nexus_dir(filename);
+            end
+            h5inst = h5info(filename, [root_dir '/instrument']);
+            dataset = read_nexus_groups_recursive(h5inst);
+            % Instrument we support must have 'moderator' and 'source' components
+            if ~any(isfield(dataset, {'moderator', 'source'}))
+                error('HERBERT:loader_nxspe:invalid_instrument', ...
+                      'nxspe file has instrument data incompatible with Horace');
+            end
+            source = IX_source(dataset.source.Name.value, '', double(dataset.source.frequency.value));
+            moderator = obj.read_inst_moderator_(dataset);
+            % NXSPE files *must* have a "fermi" component, so we distinguish
+            % instrument types based on presence of 'shaping_chopper' and 'mono_chopper'
+            % as we only have two types of instruments supported at present
+            errmsg = ['nxspe file has incomplete instrument data. Please manually set ' ...
+                       'instrument data if you want to perform resolution convolution.'];
+            if all(isfield(dataset, {'shaping_chopper', 'mono_chopper'}))
+                % The struct must have also have 'horiz_div', and 'vert_div' fields
+                if ~any(isfield(dataset, {'horiz_div', 'vert_div'}))
+                    error('HERBERT:loader_nxspe:missing_instrument_fields', errmsg);
```

Comment:
the use of errmsg here means we cannot distinguish the various errors

Reply:
Ummm I guess you're objecting to their being two error messages with the same identifier and text? (In lines 296 and 302)... what about the suggestion below?

(you comment was a bit unclear...)